src/common/abstractPromote.ts

```diff
@@ -48,20 +59,66 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
 
   protected async executePromotion(): Promise<PromotePipelineResult> {
     validateTestFlags(this.flags['test-level'], this.flags.tests);
-    const doceOrg: Org = await Org.create({ aliasOrUsername: this.flags['devops-center-username']?.getUsername() });
-    const pipelineStage: PipelineStage = await fetchAndValidatePipelineStage(
+    const doceOrg: Org = this.flags['devops-center-username'] as Org;
```

Comment:
The end result is the same, we get the DOCe org. But with less processing because we don't create an new instance of an Org, we just us the one in the flag.

Reply:
Nice!